planner/core/task.go

```diff
@@ -727,7 +737,12 @@ func finishCopTask(ctx sessionctx.Context, task task) task {
 			}
 		}
 		ts := tp.(*PhysicalTableScan)
+		prevColumnLen := len(ts.Columns)
+		prevSchema = ts.schema.Clone()
 		ts.Columns = ExpandVirtualColumn(ts.Columns, ts.schema, ts.Table.Columns)
+		if len(ts.Columns) > prevColumnLen {
+			needExtraProj = true
+		}
```

Comment:
Could you add comments to explain why we need an extra projection? How does it help to fix the panic?